public/main.js

```diff
@@ -0,0 +1,36 @@
+var calculator = {
+  add: function (x,y){
+    return x + y;
+  },
+
+  subtract: function(x,y) {
+    return x - y;
+  },
+
+  multiply: function(x,y) {
+    return x * y;
+  },
+  divide: function(x,y) {
+    if (y==0) {
+      return "Do not divide by zero."
```

Comment:
Thank you. Good Point I will make revisions.

On Thu, Mar 26, 2015 at 11:22 AM, queerviolet <notifications@github.com>
wrote:

> In public/main.js
> <https://github.com/nyc-sea-lions-2015/lightsaber/pull/70#discussion_r27222659>
> :
>
> > @@ -0,0 +1,36 @@
> > +var calculator = {
> > +  add: function (x,y){
> > +    return x + y;
> > +  },
> > +
> > +  subtract: function(x,y) {
> > +    return x - y;
> > +  },
> > +
> > +  multiply: function(x,y) {
> > +    return x * y;
> > +  },
> > +  divide: function(x,y) {
> > +    if (y==0) {
> > +      return "Do not divide by zero."
>
> It's pretty weird to have a function that sometimes returns a float and
> sometimes returns a string. I'd just return x / y and let JS division
> semantics sort it out.
>
> It turns out that dividing by zero returns Infinity:
>
> >> 2 / 0
> Infinity
>
> Which is probably what you want anyway.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nyc-sea-lions-2015/lightsaber/pull/70/files#r27222659>
> .
>